routes/users.js

```diff
@@ -1,272 +1,277 @@
-router.get('/profile', redirectLogin, async (req, res) => {
-	try {
-		res.render('profile', {
-			user: res.locals.user
-		});
-	} catch (err) {
-		res.status(400).send(JSON.stringify({
-			success: false,
-			error: err
-		}));
-	}
+router.get("/profile", redirectLogin, async (req, res) => {
+  try {
+    let user = res.locals.user;
+    let params = { user };
+    if (user.userRole == "partner") {
+      let agency = await Agency.findOne({ accountManager: user._id });
+      params = { ...params, agency };
+    }
+    res.render("profile", params);
+  } catch (err) {
+    res.status(400).send(
+      JSON.stringify({
+        success: false,
+        error: err,
+      })
+    );
+  }
 });
 
 // @desc    Update user about me info
 // @route   PUT '/users/profile'
 // @access  Private, only users
 // @tested 	No?
-router.put('/profile', redirectLogin, async (req, res) => {
-	try {
-        const {aboutMe} = req.body;
-        
-        // if no user id is present return forbidden status 403
-        if (!req.session.userId) {
-            res.status(403).send(JSON.stringify({
-                success: false,
-                error: "No user id in request"
-            }));
-        }
+router.put("/profile", redirectLogin, async (req, res) => {
+  try {
+    const { aboutMe } = req.body;
 
-        const candidate = await User.findOne({_id: req.session.userId});
+    // if no user id is present return forbidden status 403
+    if (!req.session.userId) {
+      res.status(403).send(
+        JSON.stringify({
+          success: false,
+          error: "No user id in request",
+        })
+      );
+    }
 
-        // candidate with id not found in database, return not found status 404
-        if (!candidate) {
-            res.status(404).send(JSON.stringify({
-                success: false,
-                error: "User could not be found"
-            }));
-        }
-        
-        // update user and add aboutMe
-        User.updateOne(
-            {_id: candidate._id}, 
-            {aboutMe : aboutMe },
-            {multi:true}, 
-              function(err, numberAffected){  
-              });
+    const candidate = await User.findOne({ _id: req.session.userId });
 
-        res.status(200).send(JSON.stringify({
-            success: true,
-            error: null,
-            data: aboutMe,
-        }));
+    // candidate with id not found in database, return not found status 404
+    if (!candidate) {
+      res.status(404).send(
+        JSON.stringify({
+          success: false,
+          error: "User could not be found",
+        })
+      );
+    }
 
-	} catch (err) {
-		res.status(400).send(JSON.stringify({
-			success: false,
-			error: err
-		}));
-	}
-});
+    // update user and add aboutMe
+    User.updateOne(
+      { _id: candidate._id },
+      { aboutMe: aboutMe },
+      { multi: true },
+      function (err, numberAffected) {}
```

Comment:
I think we should not ignore a failing update, it looks like even with a failed update here we return a 200? since the err is unused and will not be catched by the catch?

Reply:
Removing it since it doesn't do anything (the catch catches the error); I tried adding doing it within the function (customized message, but the catch returns the error instead).